tools/okparser/src/parser.py

```diff
@@ -0,0 +1,85 @@
+import re
+from pathlib import Path
+
+import requests
+import typer
+import yaml
+
+from tools.okparser.src import utils
+
+__REQUIRED_FIELDS__ = frozenset({"bom", "tool-list"})
+
+
+def get_wiki_data(items: list[str]) -> list[dict[str, str]]:
+    """search the wikidata api for items.
+
+    Args:
+        items: list of items(bom items, tools) to be searched.
+    Returns:
+        A list of search descriptions for each item.
+    """
+    item_list = []
+    with typer.progressbar(items, color=True) as progress:
+        for item in progress:
+            item_dict = {"identifier": "", "description": item.strip(), "link": ""}
+            # look for 'or' keyword in text and search each item
+            sub_items = item.split("or")
+            for sub_item in sub_items:
+                response = requests.get(utils.get_url(sub_item)).json()
+                if response["search"]:
+                    # get the first result from the search results for now
+                    item_dict["identifier"] = response["search"][0]["id"]
+                    item_dict["link"] = re.sub(
+                        "//www.", "https://", response["search"][0]["url"]
+                    )
+                    break
+                utils.console.print(f"\t[red]couldn't find wikidata for {sub_item}")
+            item_list.append(item_dict)
+    return item_list
+
+
+class Okh:
```

Comment:
Can we move the OKH class into a separate file?

Reply:
I was thinking if anything needed to be moved from the file, perhaps it should be the `get_wiki_data` function into utils.py rather than the OKH class since `get_wiki_data` is more of a utility function